File: Java/PastaMaker/src/test/java/pastamaker/MockDispenser.java
package pastamaker;

public class MockDispenser implements Dispenser {
	private String log = "";
	  
	public Ingredient getIngredient(IngredientType ingredient, Place place) {
		
		log += "getIngredient: ";
		log += "Ingredient Type: " + ingredient.toString();
		log += " Place: " + place.toString();
		log += "\n";
			 
		return new Ingredient(false);
	}
	
	public String toString(){
		return log;
	}

	public Ingredient getPasta(PastaType pasta, Place place) {
		log += "getPasta: ";
		log += "Pasta Type: " + pasta.toString();
		log += " Place: " + place.toString();
		log += "\n";
			 
		return new Ingredient(true);
	}

}

File: CSHARP/Refactoring/Tests/MockDispenser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RefactoringKungFu
{
    public class MockDispenser : IDispenser
    {
        private string log;
        public Ingredient GetIngredient(IngredientTypes ingredient, Places place)
        {
            log +="GetIngredient: " +
                ingredient.ToString() +
                " place: " +
                place.ToString();

            return new Ingredient(false);
        }

        public Ingredient GetPasta(PastaType pasta, Places place)
        {
            log+= "GetPlace: " +
               pasta.ToString() +
               " place: " +
               place.ToString();

            return new Ingredient(true);
        }

        public override string ToString()
        {
            return log;
        }
    }
}
